test/gleam/string_test.gleam

```diff
@@ -438,3 +438,260 @@ pub fn capitalise_test() {
+
+  string.inspect(#(1.5))
+  |> should.equal("#(1.5)")
+
+  string.inspect([#(1, 2, 3), #(1, 2, 3)])
+  |> should.equal("[#(1, 2, 3), #(1, 2, 3)]")
+
+  string.inspect(#([1, 2, 3], "🌈", #(1, "1", True)))
+  |> should.equal("#([1, 2, 3], \"🌈\", #(1, \"1\", True))")
+
+  string.inspect(Nil)
+  |> should.equal("Nil")
+
+  string.inspect(Ok(1))
+  |> should.equal("Ok(1)")
+
+  string.inspect(Ok(True))
+  |> should.equal("Ok(True)")
+
+  string.inspect(Ok(False))
+  |> should.equal("Ok(False)")
+
+  string.inspect(Ok(Nil))
+  |> should.equal("Ok(Nil)")
+
+  string.inspect(Error(2))
+  |> should.equal("Error(2)")
+
+  string.inspect(Error(True))
+  |> should.equal("Error(True)")
+
+  string.inspect(Error(False))
+  |> should.equal("Error(False)")
+
+  string.inspect(Error(Nil))
+  |> should.equal("Error(Nil)")
+
+  string.inspect(InspectTypeZero)
+  |> should.equal("InspectTypeZero")
+
+  string.inspect(InspectTypeOne(1))
+  |> should.equal("InspectTypeOne(1)")
+
+  string.inspect(InspectTypeTwo(1, 2))
+  |> should.equal("InspectTypeTwo(1, 2)")
+
+  string.inspect(InspectTypeOne([1]))
+  |> should.equal("InspectTypeOne([1])")
+
+  string.inspect(InspectTypeOne("1"))
+  |> should.equal("InspectTypeOne(\"1\")")
+
+  string.inspect(InspectTypeOne(["1"]))
+  |> should.equal("InspectTypeOne([\"1\"])")
+
+  string.inspect(InspectTypeOne(#([1], "a")))
+  |> should.equal("InspectTypeOne(#([1], \"a\"))")
+
+  string.inspect(Ok)
+  |> should.equal("//fn(a) { ... }")
+
+  string.inspect(Error)
+  |> should.equal("//fn(a) { ... }")
+
+  string.inspect(fn() { Nil })
+  |> should.equal("//fn() { ... }")
+
+  string.inspect(fn(a) {
+    a
+    Nil
+  })
+  |> should.equal("//fn(a) { ... }")
+
+  string.inspect(fn(a, b) {
+    a
+    b
+    Nil
+  })
+  |> should.equal("//fn(a, b) { ... }")
+
+  string.inspect(fn(x, y) {
+    x
+    y
+    Nil
+  })
+  |> should.equal("//fn(a, b) { ... }")
+
+  string.inspect(fn(foo: Int, bar: String) -> Bool {
+    foo
+    bar
+    False
+  })
+  |> should.equal("//fn(a, b) { ... }")
+
+  string.inspect(#(InspectTypeOne, InspectTypeTwo))
+  |> should.equal("#(//fn(a) { ... }, //fn(a, b) { ... })")
+
+  string.inspect(InspectTypeOne(InspectTypeZero))
+  |> should.equal("InspectTypeOne(InspectTypeZero)")
+}
+
+if javascript {
+  pub fn target_inspect_test() {
+    // Due to Erlang's internal representation, on Erlang this will pass, instead:
+    // |> should.equal("InspectTypeZero(InspectTypeZero)")
+    //
+    string.inspect(#(InspectTypeZero, InspectTypeZero))
+    |> should.equal("#(InspectTypeZero, InspectTypeZero)")
+
+    // Due to JavaScript's `Number` type `Float`s without digits return as `Int`s.
+    //
+    string.inspect(-1.0)
+    |> should.equal("-1")
+
+    string.inspect(0.0)
+    |> should.equal("0")
+
+    string.inspect(1.0)
+    |> should.equal("1")
+
+    string.inspect([1.0])
+    |> should.equal("[1]")
+
+    string.inspect(#(1.0))
+    |> should.equal("#(1)")
+  }
+}
+
+if erlang {
```

Comment:
Could we get some tests for some Erlang values that don't have a Gleam syntax? A pid and a ref would do. We'd need to import `erlang:self` and `erlang:make_ref` as external functions to create those data types.

Reply:
Thank you for the pointers to the erlang functions <3.

I have created these regexp by the patterns erlang returns:

1. https://regex101.com/r/gDSOmm/1 (though gleam test --target erlang always returns`erl(<0.83.0>` for me.
2. https://regex101.com/r/Xx4iuP/1